include/vigra/axistags.hxx

```diff
@@ -247,62 +263,62 @@ class AxisInfo
     {
-        return AxisInfo("z", Space, resolution, description);
+        return AxisInfo("z", Space, resolution, description, unit);
     }
 
-    static AxisInfo n(double resolution = 0.0, std::string const & description = "")
+    static AxisInfo n(double resolution = 0.0, std::string const & description = "", std::string const & unit = "")
     {
-        return AxisInfo("n", Space, resolution, description);
+        return AxisInfo("n", Space, resolution, description, unit);
     }
 
-    static AxisInfo e(double resolution = 0.0, std::string const & description = "")
+    static AxisInfo e(double resolution = 0.0, std::string const & description = "", std::string const & unit = "")
     {
-        return AxisInfo("e", Edge, resolution, description);
+        return AxisInfo("e", Edge, resolution, description, unit);
     }
 
-    static AxisInfo t(double resolution = 0.0, std::string const & description = "")
+    static AxisInfo t(double resolution = 0.0, std::string const & description = "", std::string const & unit = "")
     {
-        return AxisInfo("t", Time, resolution, description);
+        return AxisInfo("t", Time, resolution, description, unit);
     }
 
-    static AxisInfo fx(double resolution = 0.0, std::string const & description = "")
+    static AxisInfo fx(double resolution = 0.0, std::string const & description = "", std::string const & unit = "")
     {
-        return AxisInfo("x", AxisType(Space | Frequency), resolution, description);
+        return AxisInfo("x", AxisType(Space | Frequency), resolution, description, unit);
     }
 
-    static AxisInfo fy(double resolution = 0.0, std::string const & description = "")
+    static AxisInfo fy(double resolution = 0.0, std::string const & description = "", std::string const & unit = "")
     {
-        return AxisInfo("y", AxisType(Space | Frequency), resolution, description);
+        return AxisInfo("y", AxisType(Space | Frequency), resolution, description, unit);
     }
 
-    static AxisInfo fz(double resolution = 0.0, std::string const & description = "")
+    static AxisInfo fz(double resolution = 0.0, std::string const & description = "", std::string const & unit = "")
     {
-        return AxisInfo("z", AxisType(Space | Frequency), resolution, description);
+        return AxisInfo("z", AxisType(Space | Frequency), resolution, description, unit);
     }
 
-    static AxisInfo ft(double resolution = 0.0, std::string const & description = "")
+    static AxisInfo ft(double resolution = 0.0, std::string const & description = "", std::string const & unit = "")
     {
-        return AxisInfo("t", AxisType(Time | Frequency), resolution, description);
+        return AxisInfo("t", AxisType(Time | Frequency), resolution, description, unit);
     }
 
     static AxisInfo c(std::string const & description = "")
     {
         return AxisInfo("c", Channels, 0.0, description);
```

Comment:
This line instantiates channel axes with 0.0 resolution - shouldn't it also do `""` for the unit then?